File: backend/UnderSea.Bll/Dtos/User/WorldWinnerDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace UnderSea.Bll.Dtos.User
{
    public class WorldWinnerDto
    {
        public int Id { get; set; }
        public string UserName { get; set; }
        public string CountryName { get; set; }
        public int Points { get; set; }
        public int WorldRound { get; set; }
        public DateTime Date { get; set; }
    }
}

File: frontend/flutter/undersea/lib/models/response/world_winner_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'world_winner_dto.g.dart';

@JsonSerializable()
class WorldWinnerDto {
  int id;
  String? userName;
  String? countryName;
  int worldRound;
  DateTime date;
  int points;

  WorldWinnerDto(
      {required this.id,
      required this.points,
      required this.date,
      required this.worldRound,
      this.userName,
      this.countryName});

  factory WorldWinnerDto.fromJson(Map<String, dynamic> json) =>
      _$WorldWinnerDtoFromJson(json);

  /// `toJson` is the convention for a class to declare support for serialization
  /// to JSON. The implementation simply calls the private, generated
  /// helper method `_$UserToJson`.
  Map<String, dynamic> toJson() => _$WorldWinnerDtoToJson(this);
}
